crates/router/src/core/payments/operations/payment_response.rs

```diff
@@ -143,7 +150,11 @@ impl<F: Send + Clone> PostUpdateTracker<F, PaymentData<F>, types::PaymentsAuthor
             payment_data.payment_attempt.payment_method_id = payment_method_id;
             payment_data.payment_attempt.mandate_id = mandate_id;
             Ok(())
-        } else {
+        } else if save_card_flow {
```

Comment:
move it to else block , cause in some cases we just do a save card payment, without passing on_session in the request. Just customer_acceptance is required to save the card.